### DISCUSSION TITLE
SmartAutoClicker-1.3.0

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#110] [Announcements]
### New features:

- Negation operator on conditions
- Conditions can now be detected on the whole screen instead of a static position.
- Click action: You can now click on the detected condition position.
- Simplify the UI a bit.
- Debugging mode with a long press on the Play button when starting the detection.

### Bug fixes:

- https://github.com/Nain57/Smart-AutoClicker/issues/108
- https://github.com/Nain57/Smart-AutoClicker/issues/95
- Improve detection threading and stability

This update is a major one for Smart AutoClicker. I have removed my own detection engine and replaced it by OpenCv algorithms, allowing to provides new cool features such as the whole screen detection.
As the detection algorithm have change, the precision of the detection have changed as well, meaning that the threshold values for your conditions might have evolved as well. Don't worry, the update will update all your conditions threshold values to adapt them to the new behaviour.

This is still an early stage version for this new algorithm, it seems to work with my tests scenarios but I will need your feedback to improve it.

<hr /><em>This discussion was created from the release <a href='https://github.com/Nain57/Smart-AutoClicker/releases/tag/1.3.0'>SmartAutoClicker-1.3.0</a>.</em>

**Comment by kainbanford** [comment]
I'm real excited about the prospect of being able to click on the part of the screen that triggered my macro to kick off.

However, after updating to this version, I can no longer click on any part of the screen at all while the program is running.  I start a macro and get dumped to the home screen as normal, but I can't click on anything except the 4 button overlay.

Do I need to uninstall/reinstall?  When I selected the re-downloaded apk, it prompted me to update and I DO have the new feature.  (If I uninstall/reinstall do I lose previously created macros?)

Thanks!

**Reply by kainbanford** [reply]
oh, I can work around the "unable to click on anything except the floating toolbar" if I longpress play and see the percentage and then pause it again.  After that I can click on everything else just "fine"

**Reply by Nain57** [reply]
Thanks for the workaround. A bug have been opened here: https://github.com/Nain57/Smart-AutoClicker/issues/111
If you can provide your Android OS Version and phone references, it will be really helpful

**Reply by Nain57** [reply]
No need to install reinstall, this seems like an issue from the app.
I will try to fix it quickly to release a 1.3.1 without this issue

**Reply by Nain57** [reply]
I might have found the issue, but as I can't confirm myself, can anybody with this bug can check this last build ?
[smartautoclicker-debug.zip](https://github.com/Nain57/Smart-AutoClicker/files/8357553/smartautoclicker-debug.zip) 

It's a debug version, it will install alongside your release one with a different icon (blue instead of green). This allows you to test it without losing your data :)

**Reply by kainbanford** [reply]
The debug version does not have the behavior I was previously seeing!

If it still matters, I am running this on a Galaxy S20 5G (model SM-G981V) on Android 12

I do have some other unrelated questions, best to leave them here, or there's a reddit thread, or open a suggestion/wishlist?

**Reply by Nain57** [reply]
Question: you can ask here
Suggestion/wishlist: in the issue tracker: https://github.com/Nain57/Smart-AutoClicker/issues
